src/ghost-content/index.js

```diff
@@ -141,21 +143,71 @@ module.exports = ({ logger, db, projectLocation, enabled }) => {
 
   const updatePendingForAllFolders = () => Promise.each(trackedFolders, updatePendingForFolder)
 
-  const recordRevision = async (rootFolder, file, content) => {
+  const recordRevision = async (rootFolder, file, newContent) => {
     const knex = await db.get()
 
     const { normalizedFolderName } = normalizeFolder(rootFolder)
 
-    const id = await knex('ghost_content')
+    const { id, content } =
```

Comment:
This is fine logic-wise but means doing an extra read on every write attempt. Maybe it's fine but basically it means we're doing unnecessary write attempts? Maybe we can do the check on the flows module level before even calling the ghost method?

Reply:
I can move it out, but it looks to me it's good to keep this validation here so that we don't do this check for different cases. I've made query more lightweight though. What do you think?